komposable-architecture/src/main/java/com/toggl/komposable/extensions/Debounce.kt

```diff
@@ -0,0 +1,16 @@
+package com.toggl.komposable.extensions
+
+import com.toggl.komposable.architecture.Effect
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.cancellable
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.onStart
+
+fun <Action> Effect<Action>.debounce(
+    id: Any,
+    delayMillis: Long,
+): Effect<Action> = Effect {
+    this.run().debounce(delayMillis).onStart {
```

Comment:
Do we want to debounce the flow itself? 🤔 I'm not the best swift interpreter, [here's what TCA](https://github.com/pointfreeco/swift-composable-architecture/blob/dd145a13c5191634e83b8040859328bbf369f1db/Sources/ComposableArchitecture/Effects/Debounce.swift#L31) is doing.

Either way, when I remove the `.debounce(delayMillis)` the tests are still passing 🙃 

Also, we need to add some docs probably.

Reply:
Ohhhhhhh, I got it all wrong. The debounce is for the effect, not the flows! That is a bit confusing.

Reply:
The tests are still correct I believe. We are starting new effects and their starts were being "debounced". I never really emitted more than 1 item in these effects. I can add a test to make sure we that behavior doesn't change.